File: src/test/java/com/example/ex3_2_back/SecurityTest.java
package com.example.ex3_2_back;

import com.example.ex3_2_back.entity.User;
import com.example.ex3_2_back.security.MySecurity;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.Calendar;

@SpringBootTest
@Slf4j
public class SecurityTest {
    MySecurity mySecurity;

    @Autowired
    public void setMySecurity(MySecurity mySecurity) {
        this.mySecurity = mySecurity;
    }

    @Test
    void testSecurity() {
        log.info(mySecurity.genToken(new User()));
    }

    @Test
    void testGenToken() {
        var token = mySecurity.genExpiredToken(User.builder().id(1).build(), Calendar.YEAR, 100);
        log.info(token);
    }
}

File: src/test/kotlin/com/example/ex3_2_back/utiltest/SecurityTest.kt
package com.example.ex3_2_back.utiltest

import com.example.ex3_2_back.entity.User
import com.example.ex3_2_back.util.MySecurity
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.lang.Thread.sleep

@SpringBootTest
class SecurityTest {
    @Autowired
    lateinit var mySecurity: MySecurity

    val srcUser=User(id=999)

    @Test
    fun testGenToken() {
        println("GenToken")
        println(mySecurity.genToken(srcUser))
    }

    @Test
    fun testDecToken() {
        println("DecToken")
        val token = mySecurity.genToken(srcUser)
        sleep(2000)
        println(mySecurity.decToken(token))
    }
}